src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GCStaticEETypeNode.cs

```diff
@@ -112,44 +108,39 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly)
             dataBuilder.Alignment = 16;
             dataBuilder.DefinedSymbols.Add(this);
 
-            bool hasPointers = NumSeries > 0;
-            if (hasPointers)
+            int pointerSize = factory.Target.PointerSize;
+
+            int numSeries = 0;
+
+            for (int cellIndex = _gcDesc.Length - 1; cellIndex >= 0; cellIndex--)
             {
-                for (int i = ((_runLengths.Length / 2) * 2) - 1; i >= 0; i--)
+                if (_gcDesc[cellIndex])
                 {
-                    if (_targetPointerSize == 4)
-                    {
-                        dataBuilder.EmitInt(_runLengths[i]);
-                    }
-                    else
+                    numSeries++;
+
+                    int seriesSize = pointerSize;
+
+                    while (cellIndex > 0 && _gcDesc[cellIndex - 1])
                     {
-                        dataBuilder.EmitLong(_runLengths[i]);
+                        seriesSize += pointerSize;
+                        cellIndex--;
                     }
-                }
-                if (_targetPointerSize == 4)
-                {
-                    dataBuilder.EmitInt(NumSeries);
-                }
-                else
-                {
-                    dataBuilder.EmitLong(NumSeries);
+
+                    dataBuilder.EmitNaturalInt(seriesSize - _gcDesc.Length * pointerSize);
+                    dataBuilder.EmitNaturalInt(cellIndex * pointerSize);
                 }
             }
 
-            int totalSize = 0;
-            foreach (int run in _runLengths)
-            {
-                totalSize += run * _targetPointerSize;
-            }
+            Debug.Assert(numSeries > 0);
+            dataBuilder.EmitNaturalInt(numSeries);
+
+            Debug.Assert(((ISymbolNode)this).Offset == dataBuilder.CountBytes);
 
             dataBuilder.EmitShort(0); // ComponentSize is always 0
 
-            if (hasPointers)
-                dataBuilder.EmitShort(0x20); // TypeFlags.HasPointers
-            else
-                dataBuilder.EmitShort(0x00);
+            dataBuilder.EmitShort((short)(EETypeFlags.HasPointersFlag | EETypeFlags.IsInterfaceFlag));
```

Comment:
Note: I'm flipping the IsInterfaceFlag on so that `EEType::Validate` doesn't think that the EEType is bogus because it looks like an EEType for `System.Object` (no base type), but `HasReferenceFields`.

I'm starting an email thread with @PeterSolMS on how he would like to see them within the runtime (or whether we should just yank that check from `Validate`).